src/autowiring/C++11/filesystem.h

```diff
@@ -2,9 +2,13 @@
 #pragma once
 
 //C++17 Filesystem standard
-#if defined(_MSC_VER) && _MSC_VER >= 1900 //weirdly, only MSVC 2015 supports this so far.
-#include <filesystem>
-namespace awfsnamespace = std::tr2::sys;
+#if defined(_MSC_VER)
+  #include <filesystem>
+  #if _MSC_VER >= 1910
+    namespace awfsnamespace = std::experimental::filesystem;
+  #elif _MSC_VER >= 1900
+    namespace awfsnamespace = std::tr2::sys;
+  #endif
```

Comment:
I think this requires an #else clause for earlier versions of Visual Studio to use autoboost. Unless we're intentionally deprecating them in which case there should be an #error check.